dask_sql/physical/rex/core/call.py

```diff
@@ -224,14 +224,20 @@ def cast(self, operand, rex=None) -> SeriesOrScalar:
             return operand
 
         output_type = str(rex.getType())
-        output_type = sql_to_python_type(output_type.upper())
+        python_type = sql_to_python_type(output_type.upper())
 
-        return_column = cast_column_to_type(operand, output_type)
+        return_column = cast_column_to_type(operand, python_type)
 
         if return_column is None:
-            return operand
-        else:
-            return return_column
+            return_column = operand
+
+        # TODO: ideally we don't want to directly access the datetime dates,
+        # but Pandas can't truncate timezone datetimes and cuDF can't
+        # truncate datetimes
+        if output_type == "DATE":
+            return return_column.dt.strftime("%Y-%m-%d").astype(python_type)
```

Comment:
We can't cast a pandas/cudf column to a `datetime64[D]` column directly with `astype`, and we can't use `col.dt.date` to extract the dates as Python datetimes because this isn't supported by cuDF - seems like the only workaround other than that is to cast the column into a truncated "date" string column, and then cast that back to `datetime64[ns]`.

cc @shwina is there possibly a better way we could be truncating the time (hours, minutes, seconds) from a `datetime64[ns]` column?

Reply:
Opting to use the solution described [here](https://github.com/rapidsai/cudf/issues/7880#issuecomment-1030124233):

```python
In [42]: s
Out[42]:
0   2001-01-01 12:12:12
1   2001-01-02 01:02:03
dtype: datetime64[ns]

In [43]: s.dt.floor("D")
Out[43]:
0   2001-01-01
1   2001-01-02
dtype: datetime64[ns]
```